evalml/pipelines/pipeline_base.py

```diff
@@ -68,14 +67,18 @@ def __init__(
         self.random_seed = random_seed
 
         if isinstance(component_graph, list):  # Backwards compatibility
-            with warnings.catch_warnings():
-                warnings.filterwarnings(
-                    "ignore",
-                    message="ComponentGraph.from_list will be deprecated in the next release. Please use a dictionary to specify your graph instead.",
-                )
-                self.component_graph = ComponentGraph().from_list(
-                    component_graph, random_seed=self.random_seed
-                )
+            for component in component_graph:
+                component = handle_component_class(component)
+                if not component._supported_by_list_API:
+                    raise ValueError(
+                        "This component cannot be defined in a list because edges may be ambiguous. Please use a dictionary to specify the appropriate component graph for this pipeline instead."
```

Comment:
nit-pick: Should we list the component name in the message?

Reply:
Sure thing, done 😁